src/keys/public.rs

```diff
@@ -46,36 +46,30 @@ impl Public {
         encoding::encode_nano_base_32(&bits)
     }
 
-    pub fn verify(&self, message: &[u8], signature: &Signature) -> anyhow::Result<()> {
-        let result = self
-            .dalek_key()
-            .with_context(|| format!("Verify {:?} with {:?}", message, signature));
+    pub fn verify(&self, message: &[u8], signature: &Signature) -> Result<(), FeelessError> {
+        let result = self.dalek_key();
 
         match result {
-            Ok(key) => key.verify(message, &signature.internal()).with_context(|| {
```

Comment:
We've lost some context given to the user/caller. I'm not sure how to fix this? Maybe `map_err` here if possible?